app/src/main/java/org/mozilla/focus/experiments/NimbusSetup.kt

```diff
@@ -82,6 +82,7 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
                 setExperimentsLocally(R.raw.initial_experiments)
             }
 
+            context.components.settings.experimentsStartFetchingTimestamp = System.currentTimeMillis()
             // Now fetch the experiments from the server. These will be available for feature
             // configuration on the next run of the app. This call launches on the fetch thread.
             fetchExperiments()
```

Comment:
The startup sequence here still needs some work.

For first run only, when the splash screen is showing: the `NimbusInterface.Observer` should:

1. listen for the end of the fetch.
2. **Important** if and only if the `REQUEST_TIME_OUT` has not been reached, then call `applyPendingExperiments()`
3. Once `onExperimentsApplied()` is called we should arrange for the splash screen to be dismissed. 

**Important**: For subsequent runs we should not be registering an observer at all, and just calling `applyPendingExperiments()` and then `fetchExperiments()`.

Given that so much of this is tied up with the splash screen, I'd put the default `applyPendingExperiments()`/fetchExperiments()` calls here, then do the work with the observer closer to the splash screen.